### ISSUE TITLE
RemoveTask Message is not forwarded

### ISSUE THREAD

**Issue by Elfoniok** [golemfactory/clay#1942]
Currently remove task message is not forwarded. So removed task headers are still replicated on distance node. For successfully removing task headers from the network, we must do it in one broadcast.

What is more, there is a possibility that in time of propagating this message it might happen that some node already sent a task header list with removed task header to node which already processed remove task message.

**Comment by etam** [comment]
I came with this idea: Instead of actively sending `RemoveTask` message, there would be an additional field in `Tasks` message with the contents of `TaskHeaderKeeper.removed_tasks`. This way the info about removed tasks would nicely propagate throughout the network without flooding and with timeouts we already have implemented.

**Comment by Elfoniok** [comment]
Actually maybe simply propagation will work after all? No solution would work if some removed_tasks list was not present on node. Luckily taskkeeper holds such list, i didn't check the details but seems like solving problem with receiving removed tasks.
Check self.removed_tasks

**Comment by etam** [comment]
Ok, so I did it this way: if the task header is removed from `TaskKeeper`, the message is propagated. If it was removed before, it's not propagated.

The ids are removed from `removed_tasks` after 3 minutes. I wonder, if it's not too short? It is possible, that the message will propagate through the network infinitely if there's a loop where message propagates slower than this timeout. I don't know what is a reasonable timeout in such case.
